quickwit-cli/src/lib.rs

```diff
@@ -391,6 +405,53 @@ fn display_statistics(
     Ok(())
 }
 
+/// ThroughputCalculator is used to calculate throughput on time interval.
+struct ThroughputCalculator {
+    /// clock for tracking the time
+    clock: Instant,
+    /// How long it takes before calculating the throughput.
+    window_size: u64,
+    /// Stores the previous value of processed bytes.
+    processed_bytes_from_last_window: usize,
+    /// Stores the caluclated throughput.
+    throughput: f64,
+}
+
+impl ThroughputCalculator {
+    /// Creates new instance.
+    pub fn new(clock: Instant, window_size: u64) -> Self {
+        Self {
+            clock,
+            window_size,
+            processed_bytes_from_last_window: 0,
+            throughput: 0f64,
+        }
+    }
+
+    /// Calculates the throughput.
+    pub fn calculate(&mut self, processed_bytes: usize) -> f64 {
+        let elapsed_secs = self.clock.elapsed().as_secs();
```

Comment:
I was hoping for rolling windows but I guess this is ok.

Reply:
What do you mean by rolling windows?

Reply:
I think sliding window is a more common term? (or maybe they don't mean exactly the same I am unsure.)

Anyway, right now your solution updates the value at a period that is equal to the window size. That's not great for the user.
After all, the actually value is defined for every single line of logs.

```
(number of bytes(t_n)  -  number of bytes(t_n-8) / (instant(t) - instant(t_n-8)))
```